platform/settings.cpp

```diff
@@ -25,9 +25,11 @@ char const * kMeasurementUnits = "Units";
 
 StringStorage::StringStorage() : StringStorageBase(GetPlatform().SettingsPathForFile(SETTINGS_FILE_NAME)) {}
 
-StringStorage & StringStorage::Instance()
+StringStorage & StringStorage::Instance(bool reload)
 {
   static StringStorage inst;
+  if (reload)
```

Comment:
I answered this question below, to @alexzatsepin 

It's all about singletone. Someone call the `Instance()` method from the StringStorage class **before** we initialize the platform, so std::map into which the settings.ini is parsed is empty. So I decided to add the ability to reload the settings.ini on demand.